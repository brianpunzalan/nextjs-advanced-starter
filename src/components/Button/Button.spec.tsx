/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React, { ReactElement } from 'react';
import {
  create as createTestRenderer,
  ReactTestRenderer,
  ReactTestRendererJSON,
} from 'react-test-renderer';
import { ThemeProvider } from '@material-ui/styles';
import ButtonComponent from './Button';
import defaultTheme from '../../theme/default';

type JSONComponent = ReactTestRendererJSON | ReactTestRendererJSON[] | null;

describe('Button', () => {
  const StringChild = 'Click Me';
  const ChildNode = (): ReactElement => {
    return <p>Click Me</p>;
  };

  it('With Jest snapshot, <Button /> component renders with a child node', () => {
    const ButtonTestCase: ReactElement = (
      <ThemeProvider theme={defaultTheme}>
        <ButtonComponent>
          <ChildNode />
        </ButtonComponent>
      </ThemeProvider>
    );

    const createdNode: ReactTestRenderer = createTestRenderer(ButtonTestCase);
    const component: JSONComponent = createdNode.toJSON();

    expect(component).toMatchSnapshot();
  });

  it('With Jest snapshot, Button component renders with a child as string', () => {
    const ButtonTestCase: ReactElement = (
      <ThemeProvider theme={defaultTheme}>
        <ButtonComponent>{StringChild}</ButtonComponent>
      </ThemeProvider>
    );

    const createdNode: ReactTestRenderer = createTestRenderer(ButtonTestCase);
    const component: JSONComponent = createdNode.toJSON();

    expect(component).toMatchSnapshot();
  });
});
