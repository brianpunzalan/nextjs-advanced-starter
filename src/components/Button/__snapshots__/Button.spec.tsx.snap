// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button With Jest snapshot, <Button /> component renders with a child node 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-root-1 MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <p>
      Click Me
    </p>
  </span>
</button>
`;

exports[`Button With Jest snapshot, Button component renders with a child as string 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-root-1 MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Click Me
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
